src/sentry/snuba/referrer.py

```diff
@@ -887,15 +887,25 @@ class Referrer(Enum):
 
 VALUES = {referrer.value for referrer in Referrer}
 
+# These suffixes are automatically added by Query Builder code in certain conditions. Any valid referrer with these suffixes is still a valid referrer.
+VALID_SUFFIXES = ["primary", "secondary"]
+
 
 def validate_referrer(referrer: str | None) -> None:
     if not referrer:
         return
 
+    if referrer in VALUES:
+        return
+
+    for suffix in VALID_SUFFIXES:
+        if referrer.removesuffix(f".{suffix}") in VALUES:
+            return
+
     error_message = f"referrer {referrer} is not part of Referrer Enum"
+
     try:
-        if referrer not in VALUES:
-            raise Exception(error_message)
+        raise Exception(error_message)
```

Comment:
Hmm i think we can just remove the try/except here and directly call `metrics.incr` and `logger.warning`. I think you'll need to drop `exc_info` but i'm not sure

Reply:
I kept this around from the original PR because apparently this is what creates the Sentry exception, which I think the Snuba team wanted